invenio_rdm_records/services/permissions.py

```diff
@@ -63,6 +63,14 @@ class RDMRecordPermissionPolicy(RecordPermissionPolicy):
     can_authenticated = [AuthenticatedUser(), SystemProcess()]
     can_all = [AnyUser(), SystemProcess()]
 
+    #
+    # Miscellaneous
+    #
+    # Allow for querying of statistics
+    # - This is currently disabled because it's not needed and could potentially
+    #   open up surface for denial of service attacks
+    can_query_stats = [Disable()]
```

Comment:
@kpsherva the new translation layer between query API and service-layer permissions uses this new permission per default.
It should be relatively easy to adapt in a way s.t. the statistics API endpoint actually checks for record permissions based on the query parameters given by the user.

This (or a similar) permission could also be used for `stats` field-level permissions when dumping the records' stats with the marshmallow schema.